### ISSUE TITLE
Optional 2-of-2 multisig trades

### ISSUE THREAD

**Issue by Final-Phoenix** [haveno-dex/haveno#1092]
Give us 2-of-2 multisig as an option for trades.

Besides the actual XMR being traded, Alice and Bob each additionally lock up an amount of XMR equal to or greater than the value of the trade in a 2-of-2 multisig. Whichever side the scammer is on will always end up losing more than if they would have cooperated.

Not 100% guaranteed (nothing is), I know, but both sides are strongly incentivized to cooperate if they're economically rational. Vast majority of users, and even would-be scammers, are.

Pros: Doesn't potentially rely on a third party (an arbitrator), that might be in collusion, or require trusting their correct judgement in arbitrating
Cons: Have to lock up a larger proportion of XMR for trades

Unless there are good reasons this hasn't been implemented yet as an option that I just haven't thought of

**Comment by monerobull** [comment]
This has been discussed plenty and i initially was very much for this but: The main issue is placing the security deposit. If Alice sends 2 XMR of deposit to the 2:2 multisig and Bob just ghosts her, that 2 XMR is gone forever.

**Comment by niyid** [comment]
> This has been discussed plenty and i initially was very much for this but: The main issue is placing the security deposit. If Alice sends 2 XMR of deposit to the 2:2 multisig and Bob just ghosts her, that 2 XMR is gone forever.

Does a time lock expiration release the funds back to Alice?

**Comment by woodser** [comment]
>  If Alice sends 2 XMR of deposit to the 2:2 multisig and Bob just ghosts her, that 2 XMR is gone forever.

The best solution for this is coinjoin IMO, where Alice and Bob sign the same transaction to move both of their funds to multisig. This is possible but not implemented in monero-project.

**Comment by woodser** [comment]
> Does a time lock expiration release the funds back to Alice?

This isn't supported in Monero afaik.

**Comment by monerobull** [comment]
Potentially possible with FCMP transaction chaining.

**Comment by niyid** [comment]
> > Does a time lock expiration release the funds back to Alice?
> 
> This isn't supported in Monero afaik.

Yeah, we had that discussion 3/4 years ago when the project was still all talk. I remember one of you mentioning that a Feature Request will be made. Apparently nothing came of it.

**Comment by nihilist001** [comment]
related discussion on that topic : https://github.com/haveno-dex/haveno/issues/999
